pipeline/backend/types/step.go

```diff
@@ -33,4 +33,5 @@ type Step struct {
 	NetworkMode  string            `json:"network_mode,omitempty"`
 	IpcMode      string            `json:"ipc_mode,omitempty"`
 	Sysctls      map[string]string `json:"sysctls,omitempty"`
+	Resources    Resources         `json:"resources,omitempty"`
```

Comment:
I think its not the best idea to add new properties on this step level, because they are pretty backend specific. I would suggest using a more generic approach for setting backend specific options on a per step basis. We could either use the existing env vars with prefixes like `WOODPECKER_K8S_RESOURCES_REQUEST_CPU=1000m` for this or add a generic key-value map like the plugin-settings property or a property called backend with key-value string-string mappings to the step. This way other backends could also specify further per step options in the future without bloating the base schema to much.

Reply:
I prefer the plugin property settings style as well (I was simply copying the existing docker backend settings). Environment variables could still be used as an additional modifier later down the line if desired, but they're not as suitable for structured data like this.